app/views/admin/reports/_display.html.haml

```diff
@@ -0,0 +1,5 @@
+.download
+  %p.notice= t ".report_is_big"
+  %p= link_to t(".display_anyway"), file_url, target: "_blank",
+    class: "button icon icon-file",
+    onclick: "(async function (element) { element.outerHTML = await (await (await fetch(element.href))).text() })(this); return false;"
```

Comment:
Clever, a quick one-liner to load the content, with a no-JS fallback 💯 . Maybe it even falls back in the case of a `fetch` failure, which is awesome.

That's a lot of Javascript for an `onclick` handler though. It's a bit hard to see how it works, but I can see from the spec that it does :)

Perhaps it's worth a code comment to explain the intent. I think it will have to go between lines 2 & 3.
```
-# On click, the rendered report file is fetched and inserted onto the page by replacing the link.
```

Reply:
There is one too many `await` here, I did a quick test and this should be enough : 
```suggestion
    onclick: "(async function (element) { element.outerHTML = await (await fetch(element.href)).text() })(this); return false;"
```

Reply:
Ah thanks for spotting that. I committed your change, specs pass, I'll take that as approval. :wink: